lib/remote-objects.js

```diff
@@ -161,11 +161,34 @@ RemoteObjects.prototype.invoke = function(method, ctorArgs, args, callback) {
 /**
  * Get an adapter by name.
  * @param {String} name The adapter name
+ * @param {Adapter} AdapterClass the provided adapter class
  * @return {Adapter}
  */
 
-RemoteObjects.prototype.adapter = function(name) {
-  return require('./' + name + '-adapter');
+RemoteObjects.prototype.adapter = function(nameOrClass) {
+  if (typeof nameOrClass === 'function') {
+    var requiredMethods = this.getRequiredAdapterMethods();
+    var meetRequirements = this.checkAdapterRequirements(nameOrClass, requiredMethods);
+    if (!meetRequirements) {
+      throw new Error(g.f('Invalid adapter class'));
+    }
```

Comment:
I am proposing to move the validation into a new method, e.g. `this.validateAdapter(adapter)`.

Also in my experience, a generic error like "Invalid adapter class" makes it difficult for users to figure out what's the problem and how to fix it. A better solution is to change the code checking adapter requirements and return either a full error, or at least an error-message string, instead of current true/false value.